ai2thor/video_controller.py

```diff
@@ -0,0 +1,346 @@
+                              'y': r0['y'] + (ry - r0['y']) / frames * i,
+                              'z': r0['z'] + (rz - r0['z']) / frames * i},
+                    'position': {'x': p0['x'] + (px - p0['x']) / frames * i,
+                              'y': p0['y'] + (py - p0['y']) / frames * i,
+                              'z': p0['z'] + (pz - p0['z']) / frames * i},
+                    'fieldOfView': FOVstart + (FOVend - FOVstart) / frames * i,
+                    'makeAgentsVisible': visibleAgents
+                }
+                if not (smartSkybox and maxY > p0['y'] + (py - p0['y']) / frames * i):
+                    kwargs['skyboxColor'] = 'black'
+                yield self.step(**kwargs)
+            else:
+                kwargs = {
+                    'action': 'UpdateThirdPartyCamera',
+                    'thirdPartyCameraId': 0,
+                    'rotation': {'x': r0['x'] + (rx - r0['x']) / frames * i,
+                                 'y': r0['y'] + (ry - r0['y']) / frames * i,
+                                 'z': r0['z'] + (rz - r0['z']) / frames * i},
+                    'position': {'x': p0['x'] + (px - p0['x']) / frames * i,
+                                 'y': p0['y'] + (py - p0['y']) / frames * i,
+                                 'z': p0['z'] + (pz - p0['z']) / frames * i},
+                }
+                if not (smartSkybox and maxY > p0['y'] + (py - p0['y']) / frames * i):
+                    kwargs['skyboxColor'] = 'black'
+                yield self.step(**kwargs)
+
+    def lookUp(self):
+        raise NotImplementedError()
+
+    def lookDown(self):
+        raise NotImplementedError()
+
+    def stand(self):
+        """Note: have not found an easy way to move the agent in-between
+           stand and crouch."""
+        raise NotImplementedError()
+
+    def crouch(self):
+        """Note: have not found an easy way to move the agent in-between
+           stand and crouch."""
+        raise NotImplementedError()
+
+    def exportVideo(self, path):
+        """Merges all the saved frames into a .mp4 video and saves it to `path`"""
+        if self.saved_frames:
+            path = path if path[:-4] == '.mp4' else path + '.mp4'
+            if os.path.exists(path):
+                os.remove(path)
+            print((self.saved_frames[0].shape[1], self.saved_frames[0].shape[0]))
+            video = cv2.VideoWriter(
+                path,
+                cv2.VideoWriter_fourcc(*'DIVX'),
+                30,
+                (self.saved_frames[0].shape[1], self.saved_frames[0].shape[0])
+            )
+            for i, frame in enumerate(self.saved_frames):
+                print('|', end='')
+                # assumes that the frames are RGB images. CV2 uses BGR.
+                video.write(cv2.cvtColor(frame, cv2.COLOR_BGR2RGB))
+            cv2.destroyAllWindows()
+            video.release()
+            print('done')
+
+    def exportFrames(self, path):
```

Comment:
For method names that we aren't trying to match to the Unity actions it makes sense to follow the python convention of snake casing method names. So this would be ```export_frames```